site/src/modules/notifications/NotificationsInbox/NotificationsInbox.tsx

```diff
@@ -0,0 +1,111 @@
+import { getErrorDetail, getErrorMessage } from "api/errors";
+import { displayError } from "components/GlobalSnackbar/utils";
+import type { FC } from "react";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { InboxPopover } from "./InboxPopover";
+import type { Notification } from "./types";
+
+const NOTIFICATIONS_QUERY_KEY = ["notifications"];
+
+type NotificationsResponse = {
+	notifications: Notification[];
+	unread_count: number;
+};
+
+type NotificationsInboxProps = {
+	defaultOpen?: boolean;
+	fetchNotifications: () => Promise<NotificationsResponse>;
+	markAllAsRead: () => Promise<void>;
+	markNotificationAsRead: (
+		notificationId: string,
+	) => Promise<{ is_read: true }>;
```

Comment:
This is a weird type that doesn't really add any value imo. There is exactly one value which satisfies it, I'd rather it just be `void` (which would then match the `markAllAsRead` type.

```suggestion
	) => Promise<void>;
```

Reply:
It is just a placeholder until we have this type returned from the generated api types.

Reply:
I would kind of expect this api to send a 202 no content, but even if it does something, we're currently not actually using this return value is the code but it _is_ adding complexity